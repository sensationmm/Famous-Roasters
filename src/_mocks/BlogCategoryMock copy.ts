import { GraphQLError } from 'graphql'
import { loader } from 'graphql.macro'

const GET_BLOG_CATEGORY_LIST = loader('src/graphql/queries/blogCategoryListDetails.query.graphql')

export const BlogCategoryListMock = {
  request: {
    query: GET_BLOG_CATEGORY_LIST,
  },
  result: {
    data: {
      categoryList: [
        {
          __typename: 'Category',
          name: 'Kaffeewissen',
          summary: 'Werde zum Kaffee-Nerd. Unsere Experten erklären alles: Von Aroma bis Zubereitungsmethode.',
          tags: ['Nachhaltigkeit', 'SpecialityCoffee'],
        },
        {
          __typename: 'Category',
          name: 'Kaffeeequipment',
          summary:
            'Unsere Experten testen und bewerten das beste Kaffeeequipment. Wir wollen sicherstellen, dass du die beste Hardware für einen exzellenten Kaffee hast.',
          tags: ['Handfiltervergleich', 'Grundausstattung'],
        },
        {
          __typename: 'Category',
          name: 'Kaffeetrends',
          summary: 'Die aktuellsten Trends über die Kaffeewelt.',
          tags: ['Gadgets', 'Roestungen'],
        },
        {
          __typename: 'Category',
          name: 'Zubereitungstipps',
          summary:
            'Mach mehr aus deinem Kaffee: our experts prepared a few brewing guides to help you enjoy always the best coffee.',
          tags: ['Aeropress', 'Siebtraeger', 'ColdDrip'],
        },
      ],
    },
  },
}

export const BlogCategoryListMockError = {
  request: {
    query: GET_BLOG_CATEGORY_LIST,
  },
  result: {
    errors: [new GraphQLError('Error!')],
  },
}
