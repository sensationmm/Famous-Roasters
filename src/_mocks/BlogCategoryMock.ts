import { GraphQLError } from 'graphql'
import { loader } from 'graphql.macro'

const GET_BLOG_CATEGORY_LIST = loader('src/graphql/queries/blogCategoryListDetails.query.graphql')

export const BlogCategoryListMock = {
  request: {
    query: GET_BLOG_CATEGORY_LIST,
  },
  result: {
    data: {
      categoryList: [
        {
          __typename: 'Category',
          name: 'Kaffeewissen',
          summary:
            'Lever up your coffee-culture. Our expert will touch different topics: beans types, origins, cupping and many more...',
          tags: ['Nachhaltigkeit', 'SpecialityCoffee'],
        },
        {
          __typename: 'Category',
          name: 'Kaffeeequipment',
          summary:
            'Here our coffee experts test and review different equipment. They aim to cover from basic to more niche equipment.',
          tags: ['Handfiltervergleich', 'Grundausstattung'],
        },
        {
          __typename: 'Category',
          name: 'Kaffeetrends',
          summary:
            'Here our coffee experts test and review different equipment. They aim to cover from basic to more niche equipment.',
          tags: ['Gadgets', 'Roestungen'],
        },
        {
          __typename: 'Category',
          name: 'Zubereitungstipps',
          summary:
            'Mach mehr aus deinem Kaffee: our experts prepared a few brewing guides to help you enjoy always the best coffee.',
          tags: ['Aeropress', 'Siebtraeger', 'ColdDrip'],
        },
      ],
    },
  },
}

export const BlogCategoryListMockError = {
  request: {
    query: GET_BLOG_CATEGORY_LIST,
  },
  result: {
    errors: [new GraphQLError('Error!')],
  },
}
