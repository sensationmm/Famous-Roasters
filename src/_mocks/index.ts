import { CatalogueMockError, CatalogueMockMissingData, CatalogueMocks, CatalogueMocksPagination } from './CatalogueMock'
import { ProductMock } from './ProductMock'

export { CatalogueMocks as CatalogueMocks }
export { CatalogueMocksPagination as CatalogueMocksPagination }
export { CatalogueMockError as CatalogueMockError }
export { CatalogueMockMissingData as CatalogueMockMissingData }
export { ProductMock as ProductMock }
