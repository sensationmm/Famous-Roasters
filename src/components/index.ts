import { Button, ButtonColor, ButtonEmphasis, ButtonSize } from './Button'
import { Footer } from './Footer'
import { Layout } from './Layout'
import { Navigation } from './Navigation'
import { Typography, TypographySize, TypographyType } from './Typography'

export { Button as Button }
export { ButtonColor as ButtonColor }
export { ButtonEmphasis as ButtonEmphasis }
export { ButtonSize as ButtonSize }
export { Footer as Footer }
export { Layout as Layout }
export { Navigation as Navigation }
export { Typography as Typography }
export { TypographyType as TypographyType }
export { TypographySize as TypographySize }
