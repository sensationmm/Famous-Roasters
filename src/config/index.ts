import { apolloClient } from './apollo'
import i18n from './i18n'

export { i18n as i18n }
export { apolloClient as apolloClient }
