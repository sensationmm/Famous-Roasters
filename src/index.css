@tailwind base;
@tailwind components;
@tailwind utilities;

body {
  margin: 0;
  padding: 0;
  font-family: 'Open Sans', sans-serif;
  font-size: 16px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

body > div > div > main {
  min-height: 50vh;
}

:root {
  --swiper-theme-color: #646575 !important;
}

.video-container {
  position: relative;
  padding-bottom: 56.25%;
  padding-top: 0;
  height: 0;
  overflow: hidden;
}

.video-container iframe,
.video-container object,
.video-container embed {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}