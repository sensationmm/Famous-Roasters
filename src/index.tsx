import './index.css'

import React from 'react'
import { createRoot } from 'react-dom/client'

import App from './App'

const container = document.getElementById('root') as Element
const root = createRoot(container)
if (process.env.NODE_ENV === 'development') {
  root.render(
    <React.StrictMode>
      <App />
    </React.StrictMode>,
  )
} else {
  root.render(<App />)
}
