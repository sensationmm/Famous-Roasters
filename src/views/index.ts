import { About } from './About'
import { Account } from './Account'
import { Cart } from './Cart'
import { Catalogue } from './Catalogue'
import { Checkout } from './Checkout'
import { Contact } from './Contact'
import { Error } from './Error'
import { Home } from './Home'
import { OurRoasters } from './OurRoasters'
import { Product } from './Product'
import { TasteFinder } from './TasteFinder'

export { About as About }
export { Account as Account }
export { Cart as Cart }
export { Catalogue as Catalogue }
export { Checkout as Checkout }
export { Contact as Contact }
export { Error as Error }
export { Home as Home }
export { OurRoasters as OurRoasters }
export { Product as Product }
export { TasteFinder as TasteFinder }
