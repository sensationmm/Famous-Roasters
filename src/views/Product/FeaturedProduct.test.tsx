import { MockedProvider } from '@apollo/client/testing'
import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import React from 'react'
import { I18nextProvider } from 'react-i18next'
import { MemoryRouter } from 'react-router-dom'
import { ProductMock, ProductMockError, ProductMockWithCustomMetadata } from 'src/_mocks'
import { CartContext } from 'src/components'
import { i18n } from 'src/config'

import { FeaturedProduct } from '.'

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
delete window.history
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
window.history = { go: jest.fn() }
window.localStorage.setItem('tasteFinder', '')

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
  useParams: () => ({
    id: '7655228866776',
  }),
}))

describe('Featured product view', () => {
  it('Renders correctly for a successful call with aroma experimental', async () => {
    const ProductMockWithCustomMetadataOtherAroma = {
      ...ProductMockWithCustomMetadata,
      result: {
        data: {
          product: {
            ...ProductMockWithCustomMetadata.result.data.product,
            images: {
              nodes: [
                {
                  id: '123123',
                  url: 'https://cdn.shopify.com/s/files/1/0632/7251/7848/products/ezgif-4-d921ab2e2b.png?v=1649246153',
                  originalSrc: '',
                  src: '',
                  transformedSrc: '',
                },
              ],
            },
            aroma: {
              value: 'experimentell & komplex',
            },
          },
        },
      },
    }
    const { container } = render(
      <MockedProvider
        defaultOptions={{ watchQuery: { fetchPolicy: 'network-only' } }}
        mocks={[ProductMockWithCustomMetadataOtherAroma]}
        addTypename={false}
      >
        <CartContext.Provider value={{ cartId: 'gid://shopify/Cart/123456789', cartSize: 1 }}>
          <I18nextProvider i18n={i18n}>
            <MemoryRouter initialEntries={['/featured/7655228866776']}>
              <FeaturedProduct />
            </MemoryRouter>
          </I18nextProvider>
        </CartContext.Provider>
      </MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 500)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for a successful call with aroma floral', async () => {
    const ProductMockWithCustomMetadataOtherAroma = {
      ...ProductMockWithCustomMetadata,
      result: {
        data: {
          product: {
            ...ProductMockWithCustomMetadata.result.data.product,
            images: {
              nodes: [
                {
                  id: '123123',
                  url: 'https://cdn.shopify.com/s/files/1/0632/7251/7848/products/ezgif-4-d921ab2e2b.png?v=1649246153',
                  originalSrc: '',
                  src: '',
                  transformedSrc: '',
                },
              ],
            },
            aroma: {
              value: 'Floral & leicht',
            },
          },
        },
      },
    }
    const { container } = render(
      <MockedProvider
        defaultOptions={{ watchQuery: { fetchPolicy: 'network-only' } }}
        mocks={[ProductMockWithCustomMetadataOtherAroma]}
        addTypename={false}
      >
        <CartContext.Provider value={{ cartId: 'gid://shopify/Cart/123456789', cartSize: 1 }}>
          <I18nextProvider i18n={i18n}>
            <MemoryRouter initialEntries={['/featured/7655228866776']}>
              <FeaturedProduct />
            </MemoryRouter>
          </I18nextProvider>
        </CartContext.Provider>
      </MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 500)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for a successful call with aroma fruits', async () => {
    const ProductMockWithCustomMetadataOtherAroma = {
      ...ProductMockWithCustomMetadata,
      result: {
        data: {
          product: {
            ...ProductMockWithCustomMetadata.result.data.product,
            images: {
              nodes: [
                {
                  id: '123123',
                  url: 'https://cdn.shopify.com/s/files/1/0632/7251/7848/products/ezgif-4-d921ab2e2b.png?v=1649246153',
                  originalSrc: '',
                  src: '',
                  transformedSrc: '',
                },
              ],
            },
            aroma: {
              value: 'Fruchtig & lebhaft',
            },
          },
        },
      },
    }
    const { container } = render(
      <MockedProvider
        defaultOptions={{ watchQuery: { fetchPolicy: 'network-only' } }}
        mocks={[ProductMockWithCustomMetadataOtherAroma]}
        addTypename={false}
      >
        <CartContext.Provider value={{ cartId: 'gid://shopify/Cart/123456789', cartSize: 1 }}>
          <I18nextProvider i18n={i18n}>
            <MemoryRouter initialEntries={['/featured/7655228866776']}>
              <FeaturedProduct />
            </MemoryRouter>
          </I18nextProvider>
        </CartContext.Provider>
      </MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 500)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for a successful call with aroma chocolate', async () => {
    const ProductMockWithCustomMetadataOtherAroma = {
      ...ProductMockWithCustomMetadata,
      result: {
        data: {
          product: {
            ...ProductMockWithCustomMetadata.result.data.product,
            images: {
              nodes: [
                {
                  id: '123123',
                  url: 'https://cdn.shopify.com/s/files/1/0632/7251/7848/products/ezgif-4-d921ab2e2b.png?v=1649246153',
                  originalSrc: '',
                  src: '',
                  transformedSrc: '',
                },
              ],
            },
            aroma: {
              value: 'Nussig & schokoladig',
            },
          },
        },
      },
    }
    const { container } = render(
      <MockedProvider
        defaultOptions={{ watchQuery: { fetchPolicy: 'network-only' } }}
        mocks={[ProductMockWithCustomMetadataOtherAroma]}
        addTypename={false}
      >
        <CartContext.Provider value={{ cartId: 'gid://shopify/Cart/123456789', cartSize: 1 }}>
          <I18nextProvider i18n={i18n}>
            <MemoryRouter initialEntries={['/featured/7655228866776']}>
              <FeaturedProduct />
            </MemoryRouter>
          </I18nextProvider>
        </CartContext.Provider>
      </MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 500)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for a successful call with aroma spicy', async () => {
    const ProductMockWithCustomMetadataOtherAroma = {
      ...ProductMockWithCustomMetadata,
      result: {
        data: {
          product: {
            ...ProductMockWithCustomMetadata.result.data.product,
            images: {
              nodes: [
                {
                  id: '123123',
                  url: 'https://cdn.shopify.com/s/files/1/0632/7251/7848/products/ezgif-4-d921ab2e2b.png?v=1649246153',
                  originalSrc: '',
                  src: '',
                  transformedSrc: '',
                },
              ],
            },
            aroma: {
              value: 'Würzig & kräftig',
            },
          },
        },
      },
    }
    const { container } = render(
      <MockedProvider
        defaultOptions={{ watchQuery: { fetchPolicy: 'network-only' } }}
        mocks={[ProductMockWithCustomMetadataOtherAroma]}
        addTypename={false}
      >
        <CartContext.Provider value={{ cartId: 'gid://shopify/Cart/123456789', cartSize: 1 }}>
          <I18nextProvider i18n={i18n}>
            <MemoryRouter initialEntries={['/featured/7655228866776']}>
              <FeaturedProduct />
            </MemoryRouter>
          </I18nextProvider>
        </CartContext.Provider>
      </MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 500)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for a successful call without aroma', async () => {
    const ProductMockWithCustomMetadataWithoutAroma = {
      ...ProductMockWithCustomMetadata,
      result: {
        data: {
          product: {
            ...ProductMockWithCustomMetadata.result.data.product,
            images: {
              nodes: [
                {
                  id: '123123',
                  url: 'https://cdn.shopify.com/s/files/1/0632/7251/7848/products/ezgif-4-d921ab2e2b.png?v=1649246153',
                  originalSrc: '',
                  src: '',
                  transformedSrc: '',
                },
              ],
            },
            aroma: null,
          },
        },
      },
    }
    const { container } = render(
      <MockedProvider
        defaultOptions={{ watchQuery: { fetchPolicy: 'network-only' } }}
        mocks={[ProductMockWithCustomMetadataWithoutAroma]}
        addTypename={false}
      >
        <CartContext.Provider value={{ cartId: 'gid://shopify/Cart/123456789', cartSize: 1 }}>
          <I18nextProvider i18n={i18n}>
            <MemoryRouter initialEntries={['/featured/7655228866776']}>
              <FeaturedProduct />
            </MemoryRouter>
          </I18nextProvider>
        </CartContext.Provider>
      </MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 500)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for a successful call without custom metadata', async () => {
    render(
      <MockedProvider
        defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}
        mocks={[ProductMock]}
        addTypename={false}
      >
        <CartContext.Provider value={{ cartId: 'gid://shopify/Cart/123456789', cartSize: 1 }}>
          <I18nextProvider i18n={i18n}>
            <MemoryRouter initialEntries={['/featured/7655228866776']}>
              <FeaturedProduct />
            </MemoryRouter>
          </I18nextProvider>
        </CartContext.Provider>
      </MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
  })

  it('Renders correctly for an error call on product', async () => {
    render(
      <MockedProvider
        defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}
        mocks={[ProductMockError]}
        addTypename={false}
      >
        <CartContext.Provider value={{ cartId: 'gid://shopify/Cart/123456789', cartSize: 1 }}>
          <I18nextProvider i18n={i18n}>
            <MemoryRouter initialEntries={['/featured/7655228866776']}>
              <FeaturedProduct />
            </MemoryRouter>
          </I18nextProvider>
        </CartContext.Provider>
      </MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    const buttonPrompt = await screen.findByTestId('button-prompt')
    expect(buttonPrompt).toBeInTheDocument()
    fireEvent.click(buttonPrompt)
  })
})
