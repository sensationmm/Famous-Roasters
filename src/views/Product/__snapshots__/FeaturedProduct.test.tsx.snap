// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Featured product view Renders correctly for a successful call with aroma chocolate 1`] = `
<div>
  <div
    class="p-4 border border-coreUI-border bg-white"
  >
    <div
      class="text-center text-base leading-6"
    >
      Ein Fehler ist aufgetreten
    </div>
    <div
      class="mt-4 flex justify-center"
    >
      <button
        class="relative inline-flex flex-row items-center rounded-full border-2 px-4 py-2 text-base font-bold disabled:bg-brand-grey-whisper disabled:text-coreUI-text-secondary disabled:cursor-not-allowed text-brand-black bg-primary border-transparent hover:bg-brand-green-starship active:bg-primary"
        data-testid="button-prompt"
        role="button"
      >
        Erneut versuchen
      </button>
    </div>
  </div>
</div>
`;

exports[`Featured product view Renders correctly for a successful call with aroma experimental 1`] = `
<div>
  <div
    class="p-4 border border-coreUI-border bg-white"
  >
    <div
      class="text-center text-base leading-6"
    >
      Ein Fehler ist aufgetreten
    </div>
    <div
      class="mt-4 flex justify-center"
    >
      <button
        class="relative inline-flex flex-row items-center rounded-full border-2 px-4 py-2 text-base font-bold disabled:bg-brand-grey-whisper disabled:text-coreUI-text-secondary disabled:cursor-not-allowed text-brand-black bg-primary border-transparent hover:bg-brand-green-starship active:bg-primary"
        data-testid="button-prompt"
        role="button"
      >
        Erneut versuchen
      </button>
    </div>
  </div>
</div>
`;

exports[`Featured product view Renders correctly for a successful call with aroma floral 1`] = `
<div>
  <div
    class="p-4 border border-coreUI-border bg-white"
  >
    <div
      class="text-center text-base leading-6"
    >
      Ein Fehler ist aufgetreten
    </div>
    <div
      class="mt-4 flex justify-center"
    >
      <button
        class="relative inline-flex flex-row items-center rounded-full border-2 px-4 py-2 text-base font-bold disabled:bg-brand-grey-whisper disabled:text-coreUI-text-secondary disabled:cursor-not-allowed text-brand-black bg-primary border-transparent hover:bg-brand-green-starship active:bg-primary"
        data-testid="button-prompt"
        role="button"
      >
        Erneut versuchen
      </button>
    </div>
  </div>
</div>
`;

exports[`Featured product view Renders correctly for a successful call with aroma fruits 1`] = `
<div>
  <div
    class="p-4 border border-coreUI-border bg-white"
  >
    <div
      class="text-center text-base leading-6"
    >
      Ein Fehler ist aufgetreten
    </div>
    <div
      class="mt-4 flex justify-center"
    >
      <button
        class="relative inline-flex flex-row items-center rounded-full border-2 px-4 py-2 text-base font-bold disabled:bg-brand-grey-whisper disabled:text-coreUI-text-secondary disabled:cursor-not-allowed text-brand-black bg-primary border-transparent hover:bg-brand-green-starship active:bg-primary"
        data-testid="button-prompt"
        role="button"
      >
        Erneut versuchen
      </button>
    </div>
  </div>
</div>
`;

exports[`Featured product view Renders correctly for a successful call with aroma spicy 1`] = `
<div>
  <div
    class="p-4 border border-coreUI-border bg-white"
  >
    <div
      class="text-center text-base leading-6"
    >
      Ein Fehler ist aufgetreten
    </div>
    <div
      class="mt-4 flex justify-center"
    >
      <button
        class="relative inline-flex flex-row items-center rounded-full border-2 px-4 py-2 text-base font-bold disabled:bg-brand-grey-whisper disabled:text-coreUI-text-secondary disabled:cursor-not-allowed text-brand-black bg-primary border-transparent hover:bg-brand-green-starship active:bg-primary"
        data-testid="button-prompt"
        role="button"
      >
        Erneut versuchen
      </button>
    </div>
  </div>
</div>
`;

exports[`Featured product view Renders correctly for a successful call without aroma 1`] = `
<div>
  <div
    class="p-4 border border-coreUI-border bg-white"
  >
    <div
      class="text-center text-base leading-6"
    >
      Ein Fehler ist aufgetreten
    </div>
    <div
      class="mt-4 flex justify-center"
    >
      <button
        class="relative inline-flex flex-row items-center rounded-full border-2 px-4 py-2 text-base font-bold disabled:bg-brand-grey-whisper disabled:text-coreUI-text-secondary disabled:cursor-not-allowed text-brand-black bg-primary border-transparent hover:bg-brand-green-starship active:bg-primary"
        data-testid="button-prompt"
        role="button"
      >
        Erneut versuchen
      </button>
    </div>
  </div>
</div>
`;

exports[`Featured product view Renders correctly when user name is missing 1`] = `
<div>
  <div
    class="p-4 border border-coreUI-border bg-white"
  >
    <div
      class="text-center text-base leading-6"
    >
      Ein Fehler ist aufgetreten
    </div>
    <div
      class="mt-4 flex justify-center"
    >
      <button
        class="relative inline-flex flex-row items-center rounded-full border-2 px-4 py-2 text-base font-bold disabled:bg-brand-grey-whisper disabled:text-coreUI-text-secondary disabled:cursor-not-allowed text-brand-black bg-primary border-transparent hover:bg-brand-green-starship active:bg-primary"
        data-testid="button-prompt"
        role="button"
      >
        Erneut versuchen
      </button>
    </div>
  </div>
</div>
`;

exports[`Featured product view fails gracefully if no id exists 1`] = `
<div>
  <div
    class="p-4 border border-coreUI-border bg-white"
  >
    <div
      class="text-center text-base leading-6"
    >
      Ein Fehler ist aufgetreten
    </div>
    <div
      class="mt-4 flex justify-center"
    >
      <button
        class="relative inline-flex flex-row items-center rounded-full border-2 px-4 py-2 text-base font-bold disabled:bg-brand-grey-whisper disabled:text-coreUI-text-secondary disabled:cursor-not-allowed text-brand-black bg-primary border-transparent hover:bg-brand-green-starship active:bg-primary"
        data-testid="button-prompt"
        role="button"
      >
        Erneut versuchen
      </button>
    </div>
  </div>
</div>
`;

exports[`Featured product view renders as expected if no taste finder data exists 1`] = `
<div>
  <div
    class="p-4 border border-coreUI-border bg-white"
  >
    <div
      class="text-center text-base leading-6"
    >
      Ein Fehler ist aufgetreten
    </div>
    <div
      class="mt-4 flex justify-center"
    >
      <button
        class="relative inline-flex flex-row items-center rounded-full border-2 px-4 py-2 text-base font-bold disabled:bg-brand-grey-whisper disabled:text-coreUI-text-secondary disabled:cursor-not-allowed text-brand-black bg-primary border-transparent hover:bg-brand-green-starship active:bg-primary"
        data-testid="button-prompt"
        role="button"
      >
        Erneut versuchen
      </button>
    </div>
  </div>
</div>
`;
